src/st/st-theme-node.c

```diff
@@ -2644,23 +2653,35 @@ const PangoFontDescription *
 st_theme_node_get_font (StThemeNode *node)
 {
   /* Initialized despite _set flags to suppress compiler warnings */
-  PangoStyle font_style = PANGO_STYLE_NORMAL;
-  gboolean font_style_set = FALSE;
-  PangoVariant variant = PANGO_VARIANT_NORMAL;
-  gboolean variant_set = FALSE;
-  PangoWeight weight = PANGO_WEIGHT_NORMAL;
-  gboolean weight_absolute = TRUE;
-  gboolean weight_set = FALSE;
-  double size = 0.;
-  gboolean size_set = FALSE;
-
-  char *family = NULL;
+  PangoStyle font_style;
+  gboolean font_style_set;
+  PangoVariant variant;
+  gboolean variant_set;
+  PangoWeight weight;
+  gboolean weight_absolute;
+  gboolean weight_set;
+  double size;
+  gboolean size_set;
```

Comment:
What is wrong with just defining these after the return? We're sacrificing readability for less noisy compilation.

Reply:
OK.  could rework as if ..else with the declarations in the else, but the extra indention won't help readability

Reply:
I really think we should be basing these decision off of profiling the code. We're making shots in the dark here. Do we know that the early returns are the path most traveled? Otherwise, if the early return conditions are true very often, we're adding an extra step in these functions.

Reply:
sorry, don't get it.  What extra step ? I am not making any assumptions about early return path frequency.

Reply:
You're mass applying the same optimization everywhere in St. I think you are making quite an assumption because I don't think you've profiled all of these code paths. Now multiply "cheap" by 60 frames per second, that's how often these functions can get called.

Reply:
No , I haven't profiled all the paths, quite true.  That would probably take an absolute eternity, and I'm not sure I could even readily separate them all out.  I could do each change one at a time, benchmark, and see which ones make a difference.  As the code changes are so small, that would take a long time.  And I don't expect any to be negative by the nature of the change, so that would seem an awful lot of work.  The improvement is modest, about +0.9% on a quick check, and that's more than I expected as the same thing in muffin was just 0.3%. I'm quite happy grinding out this small stuff.  And I usually spot things along the way doing it, your comment elsewhere about the multiple draw_framebuffer calls has led me to another small optimisation of similar scale, so that's PRd now as well.